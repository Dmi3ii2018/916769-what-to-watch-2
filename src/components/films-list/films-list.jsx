import React from "react";
import {SmallMovieCard} from "../small-movie-card/small-movie-card";
import PropTypes from "prop-types";

export class FilmsList extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      activeCardId: -1,
    };

    this.cardOverHandler = this.cardOverHandler.bind(this);
    this.cardOutHandler = this.cardOutHandler.bind(this);
  }

  cardOverHandler(id) {
    this.setState({activeCardId: id});
  }

  cardOutHandler() {
    this.setState({activeCardId: -1});
  }

  render() {
    const {films} = this.props;

    return films.map((film) => <SmallMovieCard
      key = {`${film.id}`}
      name = {film.name}
      img = {film.img}
      id = {film.id}
      onFilmCardOver = {this.cardOverHandler}
      onFilmCardOut = {this.cardOutHandler}
    />);
  }
}

FilmsList.propTypes = {
  films: PropTypes.array,
};
