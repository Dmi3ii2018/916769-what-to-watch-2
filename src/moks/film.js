export const films = [
  {
    id: 0,
    genre: `Fantasy`,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `fantastic-beasts-the-crimes-of-grindelwald`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 1,
    genre: `Comedian`,
    name: `Bohemian Rhapsody`,
    img: `bohemian-rhapsody`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 2,
    genre: `Crime`,
    name: `Macbeth`,
    img: `macbeth`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    id: 3,
    genre: `Dramas`,
    name: `Aviator`,
    img: `aviator`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 4,
    genre: `Horror`,
    name: `We need to talk about Kevin`,
    img: `we-need-to-talk-about-kevin`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];
