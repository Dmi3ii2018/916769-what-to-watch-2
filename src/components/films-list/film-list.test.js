import React from "react";
import renderer from "react-test-renderer";
import {FilmsList} from "./films-list";
import {films} from "../../moks/film";

it(`correct render of films list`, () => {
  const tree = renderer
    .create(<FilmsList
      films = {films}
    />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
