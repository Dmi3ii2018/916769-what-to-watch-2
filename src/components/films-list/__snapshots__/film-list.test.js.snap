// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct render of films list 1`] = `ShallowWrapper {}`;
